.github/workflows/client-codegen.yaml

```diff
@@ -0,0 +1,33 @@
+name: GraphQL Client Codegen
+
+on:
+  push:
+    branches: ["**"]
```

Comment:
This workflow runs on every branch push which will execute codegen and build operations unnecessarily. This could cause resource waste and potential conflicts when multiple developers push simultaneously. Consider restricting to specific branches (like main/develop) or using pull_request events instead.